test/functional/services/lib/web_element_wrapper/web_element_wrapper.ts

```diff
@@ -687,16 +696,29 @@ export class WebElementWrapper {
    * @param {string} className
    * @return {Promise<void>}
    */
-  public async waitForDeletedByCssSelector(
-    selector: string,
-    implicitTimeout = 1000
-  ): Promise<void> {
+  public async waitForDeletedByCssSelector(selector: string, implicitTimeout = 200): Promise<void> {
     try {
+      this.logger.debug(
+        `waitForDeletedByCssSelector: implicitTimeout=${implicitTimeout}, wait=${this.timeout}`
+      );
       await this.driver.manage().setTimeouts({ implicit: implicitTimeout });
       await this.driver.wait(
         async () => {
-          const found = await this._webElement.findElements(this.By.css(selector));
-          return found.length === 0;
+          if (this.locator) {
+            try {
+              const parentElement = await this.driver.findElement(this.locator);
+              const found = await parentElement.findElements(this.By.css(selector));
+              return found.length === 0;
+            } catch (err) {
+              this.logger.debug(
+                `waitForDeletedByCssSelector: parent element is no longer in DOM, wait is done`
+              );
+              return true;
+            }
+          } else {
+            const found = await this._webElement.findElements(this.By.css(selector));
+            return found.length === 0;
+          }
```

Comment:
Combox is quite a tricky UI component and StaleElement is common case. I checked performance if we search for parent element first (it might be gone) and look for actual children only if it still there.

```
           │ debg visEditor.setInterval(Millisecond, {default, 2, false})
           │ debg comboBox.set, comboBoxSelector: visEditorInterval
           │ debg TestSubjects.find(visEditorInterval)
           │ debg Find.findByCssSelector('[data-test-subj="visEditorInterval"]') with timeout=10000
           │ debg comboBox.setElement, value: Millisecond
           │ debg comboBox.isOptionSelected, value: Millisecond
           │ debg waitForDeletedByCssSelector: implicitTimeout=50, wait=10000
           │ debg waitForDeletedByCssSelector: check count 1683729269100
           │ debg waitForDeletedByCssSelector: implicitTimeout=50, wait=10000
           │ debg waitForDeletedByCssSelector: check count 1683729269252
           │ debg TestSubjects.exists(~comboBoxOptionsList)
           │ debg Find.existsByDisplayedByCssSelector('[data-test-subj~="comboBoxOptionsList"]') with timeout=50
           │ debg Find.allByCssSelector('.euiFilterSelectItem[title^="Millisecond"]') with timeout=2500
           │ debg TestSubjects.exists(~comboBoxOptionsList)
           │ debg Find.existsByDisplayedByCssSelector('[data-test-subj~="comboBoxOptionsList"]') with timeout=50
```

It is still pretty fast, but now chances with low chances to get StaleElementReferenceException